boot/zephyr/nrf_nv_counters.c

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+#include <stdint.h>
+#include <bl_otp_counters.h>
+#include "bootutil/fault_injection_hardening.h"
+
+fih_int boot_nv_security_counter_init(void){
+    return FIH_SUCCESS;
+}
+
+/**
+ * Reads the stored value of a given image's security counter.
+ *
+ * @param image_id          Index of the image (from 0).
+ * @param security_cnt      Pointer to store the security counter value.
+ *
+ * @return                  FIH_SUCCESS on success
+ */
+fih_int boot_nv_security_counter_get(uint32_t image_id, fih_int *security_cnt){
+    int err;
+    uint16_t cur_sec_cnt;
```

Comment:
It's not clear to me why we need this, couldn't we just pass on security_cnt to get_monotonic_counter ?

Reply:
That doesn't work because of the different types.  The security_cnt variable is a fih_int and the counter variable supported by get_monotonic_counter is uint16_t. The problem as I see is that if we cast the variable in the function call the internal function will not know that it is an int variable so it will not pad the rest of the bytes with zeros. I actually tried your approach before and it failed so I will not change this.